presto-main/src/main/java/io/prestosql/sql/planner/PlanOptimizers.java

```diff
@@ -537,6 +538,21 @@ public PlanOptimizers(
                         estimatedExchangesCostCalculator,
                         ImmutableSet.of(new TransformFilteringSemiJoinToInnerJoin()))); // must run after PredicatePushDown
 
+        // Perform redirection before CBO rules to ensure stats from destination connector are used
+        // Perform redirection before agg, topN, limit, sample etc. push down into table scan as the destination connector may support a different set of push downs
+        // Perform redirection after at least one PredicatePushDown and PushPredicateIntoTableScan to allow connector to use pushed down predicates in redirection decision
+        // Perform redirection after at least table scan pruning rules because redirected table might have fewer columns
+        // PushPredicateIntoTableScan must be run after redirection
```

Comment:
it's not guaranteed, since ApplyTableScanRedirection and PushPredicateIntoTableScan are part of one IterativeOptimizer. Did you intend to have them as two separate optimizers? if not, i don't know how to interpret this comment

Reply:
As part of table scan redirection, we take predicates which were pushed down into source table scan and create a filter node on top of the target table scan node. Now we need to re-run PushPredicateIntoTableScan to allow this new filter node to be pushed down into the target table scan. This was done because target connector's predicate pushdown capability might be different to source connector's.
We are running `pushIntoTableScanOptimizer` next, so I thought that will ensure PPD into TS and satisfy the above requirement.

Reply:
i read "must be run after redirection" as "must not be run before", while you're saying "it needs to be run again afterwards"

can you please consider rewording?

Reply:
oh, i see now, ya `PushPredicateIntoTableScan needs to be run again after redirection` is more accurate, I'll update it

Reply:
created https://github.com/trinodb/trino/pull/8002
PTAL